src/50_Samples_%26_Templates/Product_Page.html

```diff
@@ -752,9 +760,33 @@ <h4 class="p2">Products that have been bought together</h4>
       </template>
     </amp-list>
     <!--
-    The button `tap` action sets the value of the `relatedProductsSrc` to an extended list of related products.
+    While infinite scrolling is not possible at the moment in AMP, you can use `amp-bind` to dynamically change the
+    src of `amp-list` and add more items to the page.
+    We bind the `src` attribute to the id of an `amp-state` component so that the `amp-list` will use the items
+    from that component as a `src`. We append items to the `amp-state` everytime the user click on the Show more button, see below for more info.
+    -->
+    <amp-state id="myState" src="/json/related_products.json">
+    </amp-state>
+    <!--
+    You can show or hide a Show more button by using `amp-form` and the `submit-error` and `submit-success` events: notice
+    how we set the `showMoreButtonDisabled` variable to false based on the server response.
     -->
-    <button [disabled]="listSrcButtonDisabled" class="ampstart-btn caps m1 mb3" id="listSrcButton" on="tap:AMP.setState({listSrcButtonDisabled: true, product: {relatedProductsSrc: '/json/more_related_products.json'}})">Show more</button>
+    <form method="GET"
+        action="/json/more_related_products_page"
+        action-xhr="/json/more_related_products_page"
+        target="_top"
+        on="submit-error: AMP.setState({
+          myState: { items: myState.items.concat(event.response.items)},
+          product: {  showMoreButtonDisabled: true }
+        }); submit-success: AMP.setState({
+          myState: { items: myState.items.concat(event.response.items)},
+          product: {moreItemsPageIndex: product.moreItemsPageIndex + 1 }
```

Comment:
just an idea: it is possible to include a boolean var `hasMorePages` in the result? We could use this to hide the show more button.

Reply:
Added, thanks